src/NzbDrone.Core/Indexers/Definitions/BinSearch.cs

```diff
@@ -165,65 +198,141 @@ public IList<ReleaseInfo> ParseResponse(IndexerResponse indexerResponse)
+                    .Where(textMatch => textMatch.Success)
+                    .Select(fileTypeMatch => fileTypeMatch.Groups["filetype"].Value);
+
+                if (fileTypes.Count() > 0)
+                {
+                    // Guess category based on filetype
+                    if (fileTypes.Any(fileType => CommonVideoFileTypes.Contains(fileType)))
+                    {
+                        release.Categories = new List<IndexerCategory> { NewznabStandardCategory.Movies, NewznabStandardCategory.TV, NewznabStandardCategory.XXX };
+                    }
+                    else if (fileTypes.Any(fileType => CommonMusicFileTypes.Contains(fileType)))
+                    {
+                        release.Categories = new List<IndexerCategory> { NewznabStandardCategory.Audio };
+                    }
+
+                    // ...
+                    else
+                    {
+                        release.Categories = new List<IndexerCategory> { NewznabStandardCategory.Other };
+                    }
+                }
+                else
+                {
+                    release.Categories = new List<IndexerCategory> { NewznabStandardCategory.Other };
+                }
+
                 releaseInfos.Add(release);
             }
 
-            return releaseInfos.ToArray();
+            return releaseInfos;
         }
 
-        private static readonly Regex ParseTitleRegex = new Regex(@"\""(?<title>.*)(?:\.(rar|nfo|mkv|par2|001|nzb|url|zip|r[0-9]{2}))\""");
-        private static readonly Regex ParseSizeRegex = new Regex(@"size: (?<size>[0-9]+(\.[0-9]+)?).(?<unit>(GB|MB|KB|B))");
-
         public Action<IDictionary<string, string>, DateTime?> CookiesUpdater { get; set; }
     }
 
+    public class BinSearchSettingsValidator : AbstractValidator<BinSearchSettings>
+    {
+        public BinSearchSettingsValidator()
+        {
+            RuleFor(c => c.BaseUrl).NotEmpty();
+
+            RuleFor(c => c.FallbackGroup).NotEmpty();
+            RuleFor(c => c.FallbackServer).NotEmpty();
+        }
+    }
+
     public class BinSearchSettings : IIndexerSettings
     {
+        private static readonly BinSearchSettingsValidator Validator = new BinSearchSettingsValidator();
+
         [FieldDefinition(1, Label = "Base Url", Type = FieldType.Select, SelectOptionsProviderAction = "getUrls", HelpText = "Select which baseurl Prowlarr will use for requests to the site")]
         public string BaseUrl { get; set; }
 
-        [FieldDefinition(2)]
+        [FieldDefinition(2, Label = "Fallback Newsgroup", Advanced = true, Type = FieldType.Textbox, HelpText = "Use this newsgroup for empty searches.")]
+        public string FallbackGroup { get; set; }
+
+        [FieldDefinition(3, Label = "Fallback Server", Advanced = true, Type = FieldType.Select, SelectOptions = typeof(BinSearchFallbackServer), HelpText = "Use this Binsearch server for empty searches.")]
```

Comment:
Is this really needed? What's the difference when selecting Primary or Secondary?

Reply:
I haven't checked all, but from comparisons I deducted that the list of indexed groups for [Primary](https://binsearch.info/groups.php) and [Secondary](https://binsearch.info/groups.php?server=2) is disjunct:
Primary "contains" the most popular groups, while Secondary "contains" the remaining ones, BinSearch indexes.